src/environmentbase/networkbase.py

```diff
@@ -347,11 +364,59 @@ def add_vpn_gateway(self,
         else:
             vpn_name = self.__class__.__name__ + 'Gateway'
 
-        gateway = self.template.add_resource(ec2.VPNGateway('vpnGateway',
+        gateway = self.add_resource(ec2.VPNGateway('vpnGateway',
             Type=vpn_conf.get('vpn_type', 'ipsec.1'),
             Tags=[ec2.Tag(key='Name', value=vpn_name)]))
 
-        gateway_connection = self.template.add_resource(ec2.VPCGatewayAttachment('vpnGatewayAttachment',
-            VpcId=self.template.vpc_id,
-            InternetGatewayId=self.template.igw,
+        gateway_connection = self.add_resource(ec2.VPCGatewayAttachment('vpnGatewayAttachment',
+            VpcId=self.vpc_id,
+            InternetGatewayId=self.igw,
             VpnGatewayId=gateway))
+
+
+class NetworkBase(EnvironmentBase):
```

Comment:
That's true, but I wanted to manage the migration from one version to the other.  I wanted to be backward compatible as much as possible. And that's also why I kept the `BaseNetwork` template in here too.